sklearn/metrics/_regression.py

```diff
@@ -1182,24 +1232,20 @@ def d2_tweedie_score(y_true, y_pred, *, sample_weight=None, power=0):
     )
     if y_type == "continuous-multioutput":
         raise ValueError("Multioutput not supported in d2_tweedie_score")
-    check_consistent_length(y_true, y_pred, sample_weight)
 
     if _num_samples(y_pred) < 2:
         msg = "D^2 score is not well-defined with less than two samples."
         warnings.warn(msg, UndefinedMetricWarning)
         return float("nan")
 
-    if sample_weight is not None:
-        sample_weight = column_or_1d(sample_weight)
-        sample_weight = sample_weight[:, np.newaxis]
-
-    dist = TweedieDistribution(power=power)
-
-    dev = dist.unit_deviance(y_true, y_pred, check_input=True)
-    numerator = np.average(dev, weights=sample_weight)
+    y_true, y_pred = np.squeeze(y_true), np.squeeze(y_pred)
+    numerator = mean_tweedie_deviance(
+        y_true, y_pred, sample_weight=sample_weight, power=power
+    )
 
-    y_avg = np.average(y_true, weights=sample_weight)
-    dev = dist.unit_deviance(y_true, y_avg, check_input=True)
-    denominator = np.average(dev, weights=sample_weight)
+    y_avg = np.full_like(y_true, np.average(y_true, weights=sample_weight))
```

Comment:
I think this is a memory regression. Previously, `y_avg` was a single value and `dist.unit_deviance` used broadcasting to compute the denominator. Now `y_avg` needs to be a full array to be passed into `mean_tweedie_deviance`.

Without changing the public API of `mean_tweedie_deviance`, a workaround is to define a `_mean_tweedie_deviance` that does not require consistent lengths and call it here and in `mean_tweedie_deviance`. Something like this: https://github.com/lorentzenchr/scikit-learn/pull/4

Reply:
Thanks for the PR!